### PR TITLE
ENH: `Styler.text_gradient`: easy extension alternative to `.background_gradient`

### PR DESCRIPTION
**Pull request by attack68**
This is minor code alteration to the `_background_gradient` function that permits a text-only coloring version, i.e. `Styler.text_gradient`.

This could alternatively be performed with a `text_only=True` keyword addition to `Styler.background_gradient` existing method, but it seemed dissimilar enough to merit its own method (and avoid adding more kwargs).

Screen Shot 2021-04-22 at 14 24 45
Screen Shot 2021-04-22 at 14 24 57

### CODE REVIEW

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -1072,6 +1072,147 @@ def hide_columns(self, subset) -> Styler:
...
+        The text color is determined according
+        to the data in each column, row or frame, or by a given
+        gradient map. Requires matplotlib.
+
+        .. versionadded:: 1.3.0
+
+        Parameters
+        ----------
+        cmap : str or colormap
+            Matplotlib colormap.
```
Comment (jreback) [inline pandas/io/formats/style.py:1098]: these match the kwargs for background_gradient? if not why not

Comment (jreback) [reply pandas/io/formats/style.py:1098]: can you use a shared doc-string?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/formats/style.py (hunk 1 above)
Comment (attack68) [reply pandas/io/formats/style.py:1098]: yes they all match its the same signature. Can you use a shared doc string even if the short summary part is different? I've never used a shared doc string.. any tips where one is used that I can lookup and copy?

Comment (attack68) [reply pandas/io/formats/style.py]: nevermind.. i figured it out and made the changes.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. can you rebase one more time and ping on green. also pls update  the styler docs as a followon.

**Comment by attack68** [comment]
@jreback pinging green now.

**Comment by simonjayhawkins** [comment]
Thanks @attack68

**Pull request merged**